src/foam/dao/daoUtils.js

```diff
@@ -113,10 +113,40 @@ foam.CLASS({
   name: 'ArraySink',
   extends: 'foam.dao.AbstractSink',
 
+  constants: {
+    // Dual to outputJSON method.
+    PARSE_JSON: function(json, opt_cls, opt_ctx) {
```

Comment:
A better name would be parseJSON, since it's really a static method rather than an actual constant.

Reply:
Re 1st comment: I would generally agree, but last I checked (which was, admittedly, a long time ago) constant definitions get CONST_CASE'd regardless of how you name them in the `constants` object. I used this naming convention for consistency between definition and invocation.

Re 2nd comment: It may be no different. I haven't used the strategy of implementing a custom `initArgs()` before. The advantage of having `parseJSON()` is the dual naming `outputJSON`/`parseJSON` in the algorithms in `JSON.js` that invoke them.

Reply:
Also, I think there is a subtle difference between `parseJSON` and `initArgs`. The former is intended to co-operate with `foam.json.parse` to *deserialize* an instance of the class. The latter is intended to *initialize* an instance of the class.

I'm adding a `TODO` to turn `PARSE_JSON` into `parseJSON` once #613 is fixed.